PopPUNK/network.py

```diff
@@ -1837,3 +1888,68 @@ def sparse_mat_to_network(sparse_mat, rlist, use_gpu = False):
                                                summarise=False)
 
     return G
+
+
+def prune_graph(prefix, reflist, passed, output_db_name, threads, use_gpu):
+    """Keep only the specified sequences in a graph
+
+    Args:
+       prefix (str)
+           Name of directory containing network
+       reflist (list)
+           Ordered list of sequences of database
+       passed (list)
+            The names of passing samples
+       output_db_name (str)
+            Name of output directory
+       threads (int)
+            Number of CPU threads to use when recalculating random match chances
+            [default = 1].
+       use_gpu (bool)
+            Whether graph is a cugraph or not
+            [default = False]
+
+    Returns:
+       vlist (list)
+           List of integers corresponding to nodes
+    """
+    if use_gpu:
+        graph_suffix = '.csv.gz'
+    else:
+        graph_suffix = '.gt'
+
+    network_found = False
+    for graph_name in ['_core.refs_graph','_core_graph','_accessory.refs_graph','_accessory_graph','.refs_graph','_graph']:
+      network_fn = f"{prefix}/{os.path.basename(prefix)}" + graph_name + graph_suffix
+      if os.path.exists(network_fn):
+          network_found = True
+          sys.stderr.write("Loading network from " + network_fn + "\n")
+          passed_set = frozenset(passed)
+          G = load_network_file(network_fn, use_gpu = use_gpu)
```

Comment:
Surely we already have a function to open a network somewhere? I think it would be better for this function to take a network rather than a prefix as input

Reply:
Agree that network interface could do with a refresh. Can we set up a new issue/branch for this (and include the GPU fast prune update from #322?